src/modules/cpu/host_tensor_augmentations.hpp

```diff
@@ -1065,4 +1065,1908 @@ RppStatus brightness_i8_i8_host_tensor(Rpp8s *srcPtr,
+                    srcPtr2Temp += 48;
+                    dstPtrTempR += 16;
+                    dstPtrTempG += 16;
+                    dstPtrTempB += 16;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount+=3)
+                {
+                    *dstPtrTempR = (Rpp8s) RPPPIXELCHECKI8((((Rpp32f) (*srcPtr1Temp) - (Rpp32f) (*srcPtr2Temp)) * alpha) + (Rpp32f) (*srcPtr2Temp));
+                    dstPtrTempR++;
+                    srcPtr1Temp++;
+                    srcPtr2Temp++;
+
+                    *dstPtrTempG = (Rpp8s) RPPPIXELCHECKI8((((Rpp32f) (*srcPtr1Temp) - (Rpp32f) (*srcPtr2Temp)) * alpha) + (Rpp32f) (*srcPtr2Temp));
+                    dstPtrTempG++;
+                    srcPtr1Temp++;
+                    srcPtr2Temp++;
+
+                    *dstPtrTempB = (Rpp8s) RPPPIXELCHECKI8((((Rpp32f) (*srcPtr1Temp) - (Rpp32f) (*srcPtr2Temp)) * alpha) + (Rpp32f) (*srcPtr2Temp));
+                    dstPtrTempB++;
+                    srcPtr1Temp++;
+                    srcPtr2Temp++;
+                }
+
+                srcPtr1Row += srcDescPtr->strides.hStride;
+                srcPtr2Row += srcDescPtr->strides.hStride;
+                dstPtrRowR += dstDescPtr->strides.hStride;
+                dstPtrRowG += dstDescPtr->strides.hStride;
+                dstPtrRowB += dstDescPtr->strides.hStride;
+            }
+        }
+
+        // Blend with fused output-layout toggle (NCHW -> NHWC)
+        else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NCHW) && (dstDescPtr->layout == RpptLayout::NHWC))
+        {
+            Rpp32u alignedLength = bufferLength & ~47;
+
+            Rpp8s *srcPtr1RowR, *srcPtr1RowG, *srcPtr1RowB, *srcPtr2RowR, *srcPtr2RowG, *srcPtr2RowB, *dstPtrRow;
+            srcPtr1RowR = srcPtr1Channel;
+            srcPtr1RowG = srcPtr1RowR + srcDescPtr->strides.cStride;
+            srcPtr1RowB = srcPtr1RowG + srcDescPtr->strides.cStride;
+            srcPtr2RowR = srcPtr2Channel;
+            srcPtr2RowG = srcPtr2RowR + srcDescPtr->strides.cStride;
+            srcPtr2RowB = srcPtr2RowG + srcDescPtr->strides.cStride;
+            dstPtrRow = dstPtrChannel;
+
+            for(int i = 0; i < roiPtr->xywhROI.roiHeight; i++)
+            {
+                Rpp8s *srcPtr1TempR, *srcPtr1TempG, *srcPtr1TempB, *srcPtr2TempR, *srcPtr2TempG, *srcPtr2TempB, *dstPtrTemp;
+                srcPtr1TempR = srcPtr1RowR;
+                srcPtr1TempG = srcPtr1RowG;
+                srcPtr1TempB = srcPtr1RowB;
+                srcPtr2TempR = srcPtr2RowR;
+                srcPtr2TempG = srcPtr2RowG;
+                srcPtr2TempB = srcPtr2RowB;
+                dstPtrTemp = dstPtrRow;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount+=16)
+                {
+                    __m128 p1[12], p2[12];
+
+                    rpp_simd_load(rpp_load48_i8pln3_to_f32pln3, srcPtr1TempR, srcPtr1TempG, srcPtr1TempB, p1);
```

Comment:
I actually discussed this with someone from our company. For U8s, the most efficient loads for PLN3 type is 16Rs, 16Gs, 16Bs since we can fit 16 U8s in one __m128i and the loads are most efficiently utilized. Then we shuffle them and write them as PKD3. It works perfectly since 16 * 3 = 48 for the loads and 12 * 4 = 48 for the stores.